### ISSUE TITLE
Convert v1 JSON test configs to v2 YAML

### ISSUE THREAD

**Issue by derekargueta** [envoyproxy/envoy#6362]
Opening this as a top-level tracking ticket for some of the work I've been doing (#6332 and #6354) of converting test configs from v1 JSON to v2 YAML.

After this ticket is completed, the only remaining uses of the v1->v2 translation code should be the `createFilterFactory` v1 interface method for filters and maybe some of the tangential tooling in `test/tools`.

A quick git grep suggests that most of the remaining tests to be updated are in the upstream health checker, logical_dns_cluster, and redis health checker. After this, I'll tackle the tooling and hopefully we can completely purge v1 code soon.

@mattklein123 feel free to assign this ticket to me.

**Comment by mattklein123** [comment]
HERO HERO HERO

**Comment by mattklein123** [comment]
@derekargueta FYI some of our coverage woes are actually due to a lot of uncovered v1 -> v2 translation code, which hopefully we can just delete when this work is done. Thank you so, so, so much for working on this. If you want some help feel free to post here with remaining things to be done and maybe we can get some people to help split it up.

**Comment by moderation** [comment]
Agree and thanks on this work @derekargueta. Based on your prior PR's I'm wondering if I could take on an _easy_ v1 to v2 migration - open to suggestions. I still have to resurrect my old v1 to v2 PR at https://github.com/envoyproxy/envoy/pull/6409/files

**Comment by derekargueta** [comment]
@mattklein123 @moderation oof, thanks for highlighting this. I'm currently working on removing `translateBootstrap` which
1) has some spots of missing code coverage
2) is the initial call to most other translation functions (i.e. `translateCdsConfig`, `translateAddress`, `translateListener`, etc.) once `translateBootstrap` is removed, we can easily parallelize removing these, but until then we still can't delete much code until `translateBootstrap` is gone. When it is, we can incrementally delete the corresponding code as we migrate v1 to v2.

Just wrestling a few last test cases in `translateBootstrap`, hopefully will have that removal up tonight or tomorrow.

Will be more mindful of how this impacts coverage.

**Comment by mattklein123** [comment]
@derekargueta no worries and thank you again for doing this! It's up to you, but if you have an idea of what needs to be done, maybe we can make a spreadsheet of all the tests that need converting and we can get people to sign up and parallelize? WDYT? If it makes sense to do that after bootstrap is done that sounds good.

**Comment by dio** [comment]
+1 for making a spreadsheet. I probably can help in common/upstream/* tests.

**Comment by derekargueta** [comment]
Started a bookkeeping spreadsheet and besides `translateCluster` (and dependents) I'm finding that we're basically blocked on deleting the `deprecated_v1` option (related #7129). Until then, we can't remove all the filter JSON translations (including HttpConnectionManager)

https://docs.google.com/spreadsheets/d/1fv7bYnkhQUYtHKJy7MgcL-vW6ZiQGpAcTjnwd5NqG7M/edit?usp=sharing

**Comment by derekargueta** [comment]
@mattklein123 we can, I'd just be concerned about a code coverage regression again.

**Comment by mattklein123** [comment]
@derekargueta IMO we should move forward. If we have to exclude the translation code from coverage we can.

**Comment by derekargueta** [comment]
@mattklein123 SGTM. In that case we can start divvying up the following
 - [ ] `translateAccessLog` (@derekargueta)
 - [ ] `translateAccessLogFilter` (@derekargueta)
 - [ ] `translateClientSslAuthFilter` (@derekargueta)
 - [ ] `translateFaultFilter`
 - [ ] `translateGrpcJsonTranscoder`
 - [ ] `translateHttpConnectionManager`
 - [ ] `translateRouter`
 - [ ] `translateSquashConfig`
 - [ ] `translateTcpProxy`
 - [ ] `translateHttp2ProtocolOptions`
 - [ ] `translateCluster`
 - [ ] `translateOutlierDetection`

^^ are called from tests, usually a `parseXyzFromJson` helper function - resolution is create a `parseXyzFromV2Yaml` function that uses `TestUtility::loadFromYaml`, convert config stubs to v2, then remove the `parseXyzFromJson` variant, then a quick `git grep` to verify no tests are calling the translate function.

Any volunteers can just mention which ones they'd like and I'll put their name next to the function.

**Comment by mattklein123** [comment]
Closing this as we can track w/ the final remaining "kill v1" issue.
